litt/README.md

```diff
@@ -0,0 +1,869 @@
+
+There are currently two implementations of the keymap in LittDB: an in-memory keymap and a keymap that uses levelDB.
+There are tradeoffs to each implementation. The in-memory keymap is faster, but has higher memory usage and longer
+startup times (it has to be rebuilt at boot time). The levelDB keymap is slower, but has a lower memory footprint and
+faster startup times.
+
+From a thread safety point of view, if a mapping is present in the keymap, the [value](#value) associated with the
+entry is guaranteed to be present on disk.
+
+- When writing a new [value](#value), it is first written to disk, and when that is complete the [key](#key) and
+  [address](#address) are written to the keymap.
+- When deleting a [value](#value), the [key](#key) and [address](#address) are first removed from the keymap, and
+  then the [value](#value) is deleted from disk.
+
+LittDB supports reading [values](#value) immediately after they are written, and during that period there may not
+be a corresponding entry in the keymap. For more information on how this edge case is handled, information about the
+[unflushed data map](#unflushed-data-map).
+
+## Read-Your-Writes Consistency
+
+The definition of read-your-writes consistency is well summarized by its name. If a thread writes a [value](#value)
+to the database and then turns around and attempts to read that [value](#value) back, it will either
+
+1. read the [value](#value) that was just written, or
+2. read an updated [value](#value) that was written AFTER the [value](#value) that was just written
+
+Note that in LittDB, values are never updated. In this context, an "updated" value the absence of a value
+when it eventually outlives its [TTL](#ttl) and is deleted by the garbage collector.
+
+An "eventual consistent" database does not necessarily provide read-your-writes consistency. In the author's experience,
+such systems can be very difficult to reason about, and can lead to subtle bugs that are difficult to track down.
+Read-your-writes consistency is simple, yet powerful and intuitive. Since providing this level of consistency
+does not hurt performance, and so the complexity of its implementation is justified.
+
+## Segment
+
+Data in LittDB [table](#table) can be visualized as a linked list. Each element in that linked list is called a
+"segment". A segment can hold many individual [values](#value). Old data is near the beginning of the list, and new
+data is near the end. Old, [expired](#ttl) data is always deleted from the first segment currently in the list. New
+data is always written to the last segment currently in the list.
+
+Segments are deleted as a whole. That is, when a segment is deleted, all data in that segment is deleted at the same
+time. Segments are only deleted when all data contained within them has [expired](#ttl).
+
+Segments have a maximum data size. When a segment is full, that segment is made immutable, and a new segment is created
+and added to the end of the list.
```

Comment:
Can you explain how segments handle storing data when the size of data to be stored will exceed the current segments max data size? Does littDB detect this, closing the current segment, and stores the data on a new (empty) segment.